iothub/service/tests/QueryTests.cs

```diff
@@ -292,5 +293,58 @@ public void QueryResultUserSuppliedContinuationTest()
             Assert.AreEqual("test", r.ElementAt(0).DeviceId);
             Assert.IsTrue(q.HasMoreResults);
         }
+
+        [TestMethod]
+        public void QueryResult_OverrideDefaultJsonSerializer_Ok()
+        {
+            // simulate json serialize/deserialize
+            var serverQueryResult = new QueryResult
+            {
+                Type = QueryResultType.Twin,
+                Items = new List<Twin>
+                {
+                    new Twin
+                    {
+                        DeviceId = "test",
+                    }
+                },
+                ContinuationToken = "GYUVJDBJFKJ"
+            };
+
+            var settings = new JsonSerializerSettings { MaxDepth = 3 };
```

Comment:
I understand the idea behind this test, but I think it might be superfluous. Since our default value is greater than 3, this test doesn't conclusively assert that the default did get overridden. 
We'd want to use a value greater than our default value in order to test this scenario.

Having said that, I'd be ok with us removing this test altogether and only testing the assertion via the "ExceedMaxDepthThrows" test case. 
The "ExceedMaxDepthThrows" test case also tests that the default is overridden, it does it with a much lighter string value, and it does that more conclusively.